src/services/SendToGovNotifyProcessor.ts

```diff
@@ -0,0 +1,54 @@
+import { Logger } from "@aws-lambda-powertools/logger";
+import { Metrics } from "@aws-lambda-powertools/metrics";
+import { SendToGovNotifyService } from "./SendToGovNotifyService";
+import { Constants } from "../utils/Constants";
+import { Email } from "../models/Email";
+import { ReminderEmail } from "../models/ReminderEmail";
+import { DynamicReminderEmail } from "../models/DynamicReminderEmail";
+import { EmailResponse } from "../models/EmailResponse";
+import { ValidationHelper } from "../utils/ValidationHelper";
+
+export class SendToGovNotifyProcessor {
+  private static instance: SendToGovNotifyProcessor;
+
+  private readonly validationHelper: ValidationHelper;
+
+  private readonly sendToGovNotifyService: SendToGovNotifyService;
+
+  constructor(private readonly logger: Logger, private readonly metrics: Metrics, YOTI_PRIVATE_KEY: string, GOVUKNOTIFY_API_KEY: string, sendToGovNotifyServiceId: string) {
+  	this.validationHelper = new ValidationHelper();
+  	this.sendToGovNotifyService = SendToGovNotifyService.getInstance(this.logger, YOTI_PRIVATE_KEY, GOVUKNOTIFY_API_KEY, sendToGovNotifyServiceId);
+  }
+
+  static getInstance(logger: Logger, metrics: Metrics, YOTI_PRIVATE_KEY: string, GOVUKNOTIFY_API_KEY: string, sendToGovNotifyServiceId: string): SendToGovNotifyProcessor {
+  	return this.instance || (this.instance = new SendToGovNotifyProcessor(logger, metrics, YOTI_PRIVATE_KEY, GOVUKNOTIFY_API_KEY, sendToGovNotifyServiceId));
+  }
+
+  async processRequest(event: any): Promise<EmailResponse | undefined> {
+  	const messageType = event.Message.messageType;
+  	const pdfPreference = event.Message.pdfPreference;
+	const postalAddress = event.Message.postalAddress;
+  	let dynamicReminderEmail: DynamicReminderEmail;
+  	let reminderEmail: ReminderEmail;
+  	let email: Email;
+
+  	switch (messageType) {
+  		case Constants.REMINDER_EMAIL_DYNAMIC:
```

Comment:
Why are we handling the reminder emails cases in here. As far as i remember it was discussed that Reminder emails will be handled by the existing GOvnotify sqs queue and this lambda should only handle the pdfPref types of emails.